### ISSUE TITLE
[BUG] stuck when analyzing "((**P) -> T) -> (**P) -> T" applying to itself

### ISSUE THREAD

**Issue by GreyElaina** [microsoft/pyright#4524]
**Describe the bug**
pyright will get stuck when analyzing a "((**P) -> T) -> ((**P) -> T)" applying to itself

**To Reproduce**
```py
from typing import Generic, TypeVar, Callable
from typing_extensions import ParamSpec

P = ParamSpec("P")
T = TypeVar("T")

def test(x: Callable[P, T]) -> Callable[P, T]:
    return x

test(lambda x: 1) # works, "(x: Unknown) -> Literal[1]" out.
reveal_type(test(test)) # stucking in this line
```

**Expected behavior**
the latest line shows "(x: (**P) -> T) -> (**P) -> T", not stucking and nothing shows.

**Screenshots or Code**
the code presented in the context.

**VS Code extension or command-line**
in a new blank github codespace, with latest version of pylance.
however, in pyright-playground(which run pyright in command-line), it stucks as vscode shows.

**Additional context**

**Comment by erictraut** [comment]
Thanks for the bug report. It will be addressed in the next release.

**Comment by erictraut** [comment]
This is addressed in pyright 1.1.292, which I just published. It will also be included in a future release of pylance.
